xds/internal/balancer/balancergroup/balancergroup.go

```diff
@@ -133,12 +140,17 @@ func (sbc *subBalancerWithConfig) stopBalancer() {
 
 type pickerState struct {
 	weight uint32
-	picker balancer.Picker
-	state  connectivity.State
+	state  balancer.State
+	// stateToAggregate is the connectivity state used only for state
+	// aggregation. It could be different from state.ConnectivityState. For
+	// example when a sub-balancer transitions from TransientFailure to
+	// connecting, state.ConnectivityState is Connecting, but stateToAggregate
+	// is still TransientFailure.
+	stateToAggregate connectivity.State
```

Comment:
Why add this new field instead of setting `pickerState.state.ConnectivityState` accordingly?